[New Rule] Netcon through XDG Autostart Entry

Aegrah opened the pull request:
## Summary
Research related to this PR is conducted and documented in https://github.com/elastic/ia-trade-team/issues/374. This issue also contains the query + validation aspect.

This PR:

- Adds a new rule to detect network connections initiated by a XDG Autostart entry on Linux XFCE/GNOME desktop distributions.

- Rule coverage will be increased when new Linux distributions are created, and this technique is tested.

No FPs last 30d my stack, no FPs last 7d in telemetry.

terrancedejesus commented:
> No FPs last 30d my stack, no FPs last 7d in telemetry.

Does this indicate we had FPs after the last 7d in telemetry? If so, is the amount extensive?

terrancedejesus commented on rules/linux/persistence_xdg_autostart_netcon.toml:
```diff
@@ -0,0 +1,100 @@
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Network connections Initiated Through XDG Autostart Entry"
```
```suggestion
name = "Network Connections Initiated Through XDG Autostart Entry"
```

terrancedejesus reviewed (approved):
Small suggestion, but LGTM!

Mikaayenson commented on rules/linux/persistence_xdg_autostart_netcon.toml:
```diff
@@ -0,0 +1,100 @@
+[rule]
+author = ["Elastic"]
+description = """
+Detects network connections initiated through XDG autostart entries for GNOME and XFCE-based Linux distributions. XDG
```
nit: 
```suggestion
Detects network connections initiated through Cross-Desktop Group (XDG) autostart entries for GNOME and XFCE-based Linux distributions. XDG
```

Mikaayenson reviewed (approved):
LGTM. Wasn't sure if you tested against the endgame data source but it may be good to double check just in case.

Aegrah commented:
@terrancedejesus No, some sequences are more compute-intense than others, so some sequences I can easily run for 14+ days, while others start taking up a lot of resources at 7+ days in telemetry. This one was tested in telemetry with 7 days of data, and locally with 30 days.

Aegrah commented:
@Mikaayenson you are right, the sensor was not running on the test VM's so therefore I did not validate. `process.entity_id` is indeed not supported by Endgame sensor, so I removed it. Ty.

Added references.
